src/main/java/com/google/sps/data/Election.java

```diff
@@ -157,29 +159,36 @@ public Election fromVoterInfoQuery(DatastoreService datastore, JSONObject voterI
       JSONArray earlyVoteSiteData = voterInfoQueryData.getJSONArray("earlyVoteSites");
       for (Object earlyVoteSite : earlyVoteSiteData) {
         JSONObject earlyVoteSiteJSON = (JSONObject) earlyVoteSite;
-        pollingStations.add(createPollingStation(earlyVoteSiteJSON, "earlyVoteSite"));
+        pollingStations.add(createPollingStation(earlyVoteSiteJSON, "earlyVoteSite", datastore));
       }
     }
 
     if (voterInfoQueryData.has("dropOffLocations")) {
       JSONArray dropOffData = voterInfoQueryData.getJSONArray("dropOffLocations");
       for (Object dropOff : dropOffData) {
         JSONObject dropOffJSON = (JSONObject) dropOff;
-        pollingStations.add(createPollingStation(dropOffJSON, "dropOffLocation"));
+        pollingStations.add(createPollingStation(dropOffJSON, "dropOffLocation", datastore));
       }
     }
 
     if (voterInfoQueryData.has("pollingLocations")) {
       JSONArray pollingLocationData = voterInfoQueryData.getJSONArray("pollingLocations");
       for (Object pollingLocation : pollingLocationData) {
         JSONObject pollingLocationJSON = (JSONObject) pollingLocation;
-        pollingStations.add(createPollingStation(pollingLocationJSON, "pollingLocation"));
+        pollingStations.add(
+            createPollingStation(pollingLocationJSON, "pollingLocation", datastore));
       }
     }
 
+    if (pollingStations.size() > 0) {
+      pollingStationSet = ImmutableSet.copyOf((Collection<EmbeddedEntity>) pollingStations);
```

Comment:
Do you need this if condition? a copy of an empty set is still just an empty set, which is what the else is doing anything.

Reply:
You're correct -- I was using this line of code for debugging purposes and will delete it now.